File: Challenges/GuessTheNumber/Samples/CS/GuessTheNumberImplementation/NumberGuesser.cs
using System;
using System.Collections.Generic;
using System.Linq;
using OmahaMTG.Challenge.GuessTheNumberChallenge;

namespace GuessTheNumberImplementation
{
    public class NumberGuesser : IGuessTheNumberChallenge
    {
        public string AuthorNotes
        {
            get { return string.Empty; }
        }

        public int GuessNumber(IEnumerable<int> sequence, Func<int, bool> checkAnswer)
        {
            int answer = 0;

            for (int i = 0; i <= 20; i++)
                if (checkAnswer(answer))
                    break;
                else
                    answer += 1;

            return answer;
        }

    }

}

File: Challenges/GuessTheNumber/Samples/VB/GuessTheNumberImplementation/NumberGuesser.vb
Imports OmahaMTG.Challenge.ChallengeCommon
Imports OmahaMTG.Challenge.GuessTheNumberChallenge

Public Class NumberGuesser
    Implements IGuessTheNumberChallenge

    Public ReadOnly Property AuthorNotes() As String Implements IChallenge.AuthorNotes
        Get
            Return String.Empty
        End Get
    End Property

    Public Function GuessNumber(sequence As IEnumerable(Of Integer), checkAnswer As Func(Of Integer, Boolean)) As Integer Implements IGuessTheNumberChallenge.GuessNumber
        Dim answer As Integer = 0

        For i As Integer = 0 To 20
            If checkAnswer(answer) Then
                Exit For
            End If
            answer += 1
        Next

        Return answer
    End Function

End Class
